### PR TITLE
Timer

### PR DESCRIPTION
**Pull request by danimoh**
Requires https://github.com/nimiq/nimiq-utils/pull/27 to get merged and deployed first.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
             ]
         },
         "lib": [
-            "es2015",
+            "es2017",
```
[the remark this answers is not in the document]
Comment (danimoh) [reply tsconfig.json]: Yes, to be able to use `Array.includes`. `es2016` would suffice for that, but I think at this point we can also already upgrade to `es2017`.

File: package.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
   },
   "devDependencies": {
     "@nimiq/iqons": "^1.1.1",
-    "@nimiq/utils": "^0.4.1",
+    "@nimiq/utils": "link:../nimiq.github.io/libraries/nimiq-utils",
```
[the remark this answers is not in the document]
Comment (danimoh) [reply package.json]: Cf. https://github.com/nimiq/vue-components/pull/49#discussion_r337992149

File: tsconfig.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (sisou) [reply tsconfig.json]: I looked it up. The `lib` is the one allowed in the src, and the `target` is the version that ts compiles to.

So I think the correct approach is to put `esnext` as the lib (if that's possible) and put `es2015`=`es6` as the `target`.

File: src/components/Timer.vue [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+        const minInterval = 1000 / 60; // up to 60 fps
+        const maxInterval = this.detailsShown && this._timeLeft < 60000
+            ? 500 // when counting down seconds update more regularly
+            : Number.POSITIVE_INFINITY;
+        return Math.max(minInterval, Math.min(maxInterval, this._totalTime / steps));
+    }
+
+    @Watch('detailsShown', { immediate: true })
+    private _setRadius() {
+        this.radius.tweenTo(this.detailsShown ? 12 : 8, 300);
```
Comment (nibhar) [inline src/components/Timer.vue]: Do these numbers correspond to something? If so can be added as comment to the constants.

```ts
private static readonly RADIUS_BASE = 8;
private static readonly RADIUS_GROW_FACTOR = 1.5;
```
```suggestion
        this.radius.tweenTo(Timer.RADIUS_BASE * (this.detailsShown ? Timer.RADIUS_GROW_FACTOR : 1), 300);
```

**Pull request merged**